pkg/plugins/datasource_plugin.go

```diff
@@ -107,13 +112,16 @@ func (p *DataSourcePlugin) restartKilledProcess(ctx context.Context) error {
 		case <-ctx.Done():
 			return ctx.Err()
 		case <-ticker.C:
-			if p.client.Exited() {
-				err := p.spawnSubProcess()
-				p.log.Debug("Spawning new sub process", "name", p.Name, "id", p.Id)
-				if err != nil {
-					p.log.Error("Failed to spawn subprocess")
-				}
+			if !p.client.Exited() {
+				continue
 			}
+
+			if err := p.spawnSubProcess(); err != nil {
+				p.log.Error("Failed to spawn subprocess")
+				continue
+			}
+
+			p.log.Debug("Spawned new sub process", "name", p.Name, "id", p.Id)
```

Comment:
Removed arguments since logger already includes plugin id, see https://github.com/grafana/grafana/blob/master/pkg/plugins/datasource_plugin.go#L61.

```suggestion
			p.log.Debug("Plugin process restarted")
```